## Task 5 — Country Revenue Analysis

### Business Question

Which countries generate the highest total monthly revenue?

### Analysis

Group customers by country, calculate the total monthly revenue for each country, sort the results by total revenue in descending order, and identify the countries with the highest revenue.

In [3]:
import pandas as pd
import numpy as np
customers = pd.read_csv("starlink_customers.csv")
subscriptions = pd.read_csv("subscriptions.csv")
starlink = customers.merge(
    subscriptions,
    left_on = "unique_id",
    right_on = "customer_id",
    how = "left"
)
total_revenue_by_country = starlink.groupby("country").agg(
    total_revenue = ("monthly_fee", "sum")
).sort_values(
    "total_revenue",
    ascending = False
)
print(total_revenue_by_country)

                total_revenue
country                      
Australia            122360.0
Spain                122100.0
Ukraine              121760.0
Canada               119630.0
Germany              118550.0
France               117780.0
Japan                117710.0
Brazil               115680.0
United States        114250.0
United Kingdom       113450.0


### Result

The analysis shows that Australia generates the highest total monthly revenue at 122,360, followed by Spain with 122,100 and Ukraine with 121,760. Canada and Germany complete the top five countries by monthly revenue.

### Business Insight

Australia generates the highest monthly revenue among all countries, with 122,360, while Spain and Ukraine show very similar revenue levels. The relatively small difference between the top countries suggests that monthly revenue is distributed fairly evenly across the main markets.